### PR TITLE
chore: Migrated deprecated apis to iOS 15 version - WPB-5015

### PR DESCRIPTION
**Pull request by theswiftway**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5015" title="WPB-5015" target="_blank">StoryWPB-5015</a>  [iOS]  Replace deprecated in iOS 15.0 methods
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_A lot of deprecation warnings due to migration to iOS 15._

### Causes (Optional)

_Migration to iOS 15._

### Solutions

_Migrating old APIs to use newer APIs_

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: wire-ios-request-strategy/Sources/Notifications/Push Notifications/Notification Types/Content/ZMLocalNotification+Events.swift [hunk 1]
```diff
@@ -184,10 +184,6 @@ private class ConversationCreateEventNotificationBuilder: EventNotificationBuild
         return LocalNotificationType.event(.conversationCreated)
     }
 
-    override func shouldCreateNotification() -> Bool {
```
Comment (KaterinaWire) [inline wire-ios-request-strategy/Sources/Notifications/Push Notifications/Notification Types/Content/ZMLocalNotification+Events.swift:187]: why did you remove this?

File: wire-ios-sync-engine/Source/UserSession/ZMUserSession+Push.swift [hunk 2]
```diff
@@ -200,7 +200,7 @@ extension ZMUserSession: UNUserNotificationCenterDelegate {
             let shouldPresent = responder?.shouldPresentNotification(with: userInfo)
 
             var options = UNNotificationPresentationOptions()
-            if shouldPresent ?? true { options = [.alert, .sound] }
```
Comment (KaterinaWire) [inline wire-ios-sync-engine/Source/UserSession/ZMUserSession+Push.swift:203]: is it also deprecated?

Comment (theswiftway) [reply wire-ios-sync-engine/Source/UserSession/ZMUserSession+Push.swift:203]: Yes. This is also deprecated

File: wire-ios-request-strategy/Sources/Notifications/Push Notifications/Notification Types/Content/ZMLocalNotification+Events.swift (hunk 1 above)
Comment (theswiftway) [reply wire-ios-request-strategy/Sources/Notifications/Push Notifications/Notification Types/Content/ZMLocalNotification+Events.swift:187]: clean up. It isn't doing anything

**Comment by theswiftway** [comment]
There are changes w.r.t iOS 15 for UIButton's appearance, which isn't in the scope for this PR.

File: wire-ios/Wire-iOS Share Extension/PasscodeTextField.swift [hunk 3]
```diff
@@ -45,9 +45,12 @@ class PasscodeTextField: AccessoryTextField {
 
         iconButton.tintColor = UIColor.Team.textColor
         iconButton.setBackgroundImage(UIImage.singlePixelImage(with: .clear), for: state)
-
-        iconButton.adjustsImageWhenDisabled = false
-
+        iconButton.configurationUpdateHandler = { button in
+            switch button.state {
+            default:
+                button.imageView?.tintAdjustmentMode = .normal
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS Share Extension/PasscodeTextField.swift:51]: Why did you implement these changes only for `disabled` state in the `ValidatedTextField.swift` ? Shouldn't be the same?

File: wire-ios-sync-engine/Source/Calling/CallKitManager.swift [hunk 4]
```diff
@@ -162,9 +162,7 @@ public class CallKitManager: NSObject, CallKitManagerInterface {
     }
 
     static var providerConfiguration: CXProviderConfiguration {
-        let localizedName = Bundle.main.infoDictionary?["CFBundleName"] as? String ?? "Wire"
-        let configuration = CXProviderConfiguration(localizedName: localizedName)
-
+        let configuration = CXProviderConfiguration()
```
Comment (KaterinaWire) [inline wire-ios-sync-engine/Source/Calling/CallKitManager.swift:165]: do we have a workaround in iOS 15 for a custom name?

File: wire-ios-sync-engine/Source/Calling/AVS/AVSWrapper.swift [hunk 5]
```diff
@@ -168,9 +168,7 @@ public class AVSWrapper: AVSWrapperType {
             buffer = Data(count: 0)
         }
 
-        buffer.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) in
-            wcall_sft_resp(handle, error, bytes, buffer.count, context)
-        }
+        buffer.withUnsafeBytes { wcall_sft_resp(handle, error, $0.bindMemory(to: UInt8.self).baseAddress!, buffer.count, context)}
```
Comment (KaterinaWire) [inline wire-ios-sync-engine/Source/Calling/AVS/AVSWrapper.swift]: maybe we should avoid forced unwrapping

File: wire-ios-utilities/Source/Validation/ZMEmailAddressValidator.swift [hunk 6]
```diff
@@ -88,16 +88,11 @@ import UIKit
         let emailScanner = Scanner(string: emailAddress! as String)
         emailScanner.charactersToBeSkipped = CharacterSet()
         emailScanner.locale = Locale(identifier: "en_US_POSIX")
+        let local: String? =  emailScanner.scanUpToString("@")
```
Comment (KaterinaWire) [inline wire-ios-utilities/Source/Validation/ZMEmailAddressValidator.swift]:
```suggestion
        let local: String? = emailScanner.scanUpToString("@")
```

File: wire-ios-utilities/Source/Validation/ZMEmailAddressValidator.swift [hunk 7]
```diff
@@ -88,16 +88,11 @@ import UIKit
...
+        _ = emailScanner.scanString("@")
```
Comment (KaterinaWire) [inline wire-ios-utilities/Source/Validation/ZMEmailAddressValidator.swift:92]: why do we need this?

Comment (theswiftway) [reply wire-ios-utilities/Source/Validation/ZMEmailAddressValidator.swift:92]: to  move the scanner beyond `@`

File: wire-ios-sync-engine/Source/Calling/CallKitManager.swift (hunk 4 above)
Comment (theswiftway) [reply wire-ios-sync-engine/Source/Calling/CallKitManager.swift:165]: No. that has been deprecated

File: wire-ios-utilities/Source/Validation/ZMEmailAddressValidator.swift (hunk 7 above)
Comment (theswiftway) [reply wire-ios-utilities/Source/Validation/ZMEmailAddressValidator.swift:92]: We should have regex here. Not sure why this is done in this specific way. Doing that isn't in the scope of this PR.

File: wire-ios/Wire-iOS Share Extension/PasscodeTextField.swift (hunk 3 above)
Comment (theswiftway) [reply wire-ios/Wire-iOS Share Extension/PasscodeTextField.swift:51]: Good catch! This should be done only when it is  disabled.

**Review by KaterinaWire — APPROVED** [review]
